.github/workflows/binaries.yml

```diff
@@ -32,49 +32,61 @@ jobs:
             target: x86_64-unknown-linux-gnu
           - os: ubuntu-22.04-arm # Use 22 to get an older version of glibc for increased compat
             target: aarch64-unknown-linux-gnu
+    runs-on: ${{ matrix.sys.os }}
+    steps:
+      - uses: actions/checkout@v5
+      - uses: ./.github/actions/build-binary
+        with:
+          target: ${{ matrix.sys.target }}
+          crate-name: ${{ matrix.crate.name }}
+          binary-name: ${{ matrix.crate.binary }}
+
+  build-macos:
+    if: github.event_name == 'release' || (github.event_name == 'push' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/release/')))
+    strategy:
+      fail-fast: false
+      matrix:
+        crate:
+          - name: stellar-cli
+            binary: stellar
+        sys:
           - os: macos-14
             target: aarch64-apple-darwin
```

Comment:
Heads up if you wanted to keep the build binary workflow inside the binaries.yml, another way would be to include a new parameter here in the matrix for the macos and windos builds, something like "build-on-prs: false" and then on the linux ones have "build-on-prs: true", and then filter within the existing workflow with that.